syntax = "proto3";

package anki.stats;

message RevlogEntry {
  enum ReviewKind {
    LEARNING = 0;
    REVIEW = 1;
    RELEARNING = 2;
    FILTERED = 3;
    MANUAL = 4;
    RESCHEDULED = 5;
  }
  int64 id = 1;
  int64 cid = 2;
  int32 usn = 3;
  uint32 button_chosen = 4;
  int32 interval = 5;
  int32 last_interval = 6;
  uint32 ease_factor = 7;
  uint32 taken_millis = 8;
  ReviewKind review_kind = 9;
}

message CardEntry {
  int64 id = 1;
  int64 note_id = 2;
  int64 deck_id = 3;
}

message DeckEntry {
  int64 id = 1;
  int64 parent_id = 2;
  int64 preset_id = 3;
}

message Dataset {
  repeated RevlogEntry revlogs = 1;
  repeated CardEntry cards = 2;
  repeated DeckEntry decks = 3;
  int64 next_day_at = 4;
}
